src/handwritten/system/_collections/physical_channel_collection.py

```diff
@@ -59,9 +59,21 @@ def __getitem__(self, index):
         if isinstance(index, int):
             return _PhysicalChannelAlternateConstructor(self.channel_names[index], self._interpreter)
         elif isinstance(index, slice):
-            return _PhysicalChannelAlternateConstructor(self.channel_names[index], self._interpreter)
+            return [_PhysicalChannelAlternateConstructor(channel, self._interpreter) for channel in self.channel_names[index]]
         elif isinstance(index, str):
-            return _PhysicalChannelAlternateConstructor(f'{self._name}/{index}', self._interpreter)
+            requested_channels = unflatten_channel_string(index)
+            # Ensure the channel names are fully qualified. If the channel is invalid, the user will get errors from the
+            # channel objects on use.
+            channels_to_use = []
+            for channel in requested_channels:
+                if channel.startswith(self._name):
+                    channels_to_use.append(channel)
```

Comment:
`self._name` may be a prefix of the requested device name. 

Example:
`Device("Dev1").ai_physical_chans["Dev100/ai0"]`

 I think it's better for this case to error, even if the error message isn't great.
```suggestion
                if channel.startswith(f"{self._name}/"):
                    channels_to_use.append(channel)
```